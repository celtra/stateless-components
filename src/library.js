import '@/stateless/define_helpers'

import * as filtersObject from '@/helpers/filters.js'
import * as directivesObject from '@/helpers/directives.js'
import * as itemsUtilsObject from '@/stateless/items_utils.js'

import CheckboxComponent from '@/stateless/checkbox.vue'
import DialogButtonComponent from '@/stateless/dialog_button.vue'
import DialogComponent from '@/stateless/dialog.vue'
import DropAreaComponent from '@/stateless/drop_area.vue'
import FileUploadRequirementsComponent from '@/stateless/file_upload_requirements.vue'
import FileUploadComponent from '@/stateless/file_upload.vue'
import ImageListComponent from '@/stateless/image_list.vue'
import InputComponent from '@/stateless/input.vue'
import MultiselectComponent from '@/stateless/multiselect.vue'
import RadioButtonComponent from '@/stateless/radiobutton.vue'
import SelectboxComponent from '@/stateless/selectbox.vue'
import SupportTextComponent from '@/stateless/support_text.vue'
import IconComponent from '@/stateless/icon.vue'
import ToastComponent from '@/stateless/Toast.vue'
import PieChartComponent from '@/stateless/PieChart.vue'
import InlineDialogComponent from '@/stateless/InlineDialog.vue'
import DefaultListComponent from '@/stateless/DefaultList.vue'
import DefaultListItemComponent from '@/stateless/DefaultListItem.vue'
import ScrollableListComponent from '@/stateless/ScrollableList.vue'
import TypeaheadComponent from '@/stateless/Typeahead.vue'
import TypeaheadMultiselectComponent from '@/stateless/TypeaheadMultiselect.vue'
import TooltipComponent from '@/stateless/Tooltip.vue'

export const filters = filtersObject
export const directives = directivesObject
export const itemsUtils = itemsUtilsObject

export const Checkbox = CheckboxComponent
export const DialogButton = DialogButtonComponent
export const Dialog = DialogComponent
export const DropArea = DropAreaComponent
export const FileUploadRequirements = FileUploadRequirementsComponent
export const FileUpload = FileUploadComponent
export const ImageList = ImageListComponent
export const Input = InputComponent
export const Multiselect = MultiselectComponent
export const RadioButton = RadioButtonComponent
export const Selectbox = SelectboxComponent
export const SupportText = SupportTextComponent
export const Icon = IconComponent
export const Toast = ToastComponent
export const PieChart = PieChartComponent
export const InlineDialog = InlineDialogComponent
export const DefaultList = DefaultListComponent
export const DefaultListItem = DefaultListItemComponent
export const ScrollableList = ScrollableListComponent
export const Typeahead = TypeaheadComponent
export const TypeaheadMultiselect = TypeaheadMultiselectComponent
export const Tooltip = TooltipComponent
