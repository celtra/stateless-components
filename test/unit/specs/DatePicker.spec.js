import { DatePicker } from '@/library.js'
import snapshotMixin from '../component_snapshot_mixin'

describe('DatePicker', () => {
    snapshotMixin(DatePicker)
})
