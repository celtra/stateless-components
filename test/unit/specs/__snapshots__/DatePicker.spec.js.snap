// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker computed usecase-0__disabled-false caretSize 1`] = `"12px"`;

exports[`DatePicker computed usecase-0__disabled-false formattedDate 1`] = `"2018-01-10T00:00:00+00:00"`;

exports[`DatePicker computed usecase-0__disabled-false isEmpty 1`] = `false`;

exports[`DatePicker computed usecase-0__disabled-false states 1`] = `
Object {
  "disabled": false,
  "error": undefined,
  "valid": undefined,
}
`;

exports[`DatePicker computed usecase-0__disabled-true caretSize 1`] = `"12px"`;

exports[`DatePicker computed usecase-0__disabled-true formattedDate 1`] = `"2018-01-10T00:00:00+00:00"`;

exports[`DatePicker computed usecase-0__disabled-true isEmpty 1`] = `false`;

exports[`DatePicker computed usecase-0__disabled-true states 1`] = `
Object {
  "disabled": true,
  "error": undefined,
  "valid": undefined,
}
`;

exports[`DatePicker computed usecase-1__disabled-false caretSize 1`] = `"12px"`;

exports[`DatePicker computed usecase-1__disabled-false formattedDate 1`] = `"2018-01-10T00:00:00+00:00"`;

exports[`DatePicker computed usecase-1__disabled-false isEmpty 1`] = `false`;

exports[`DatePicker computed usecase-1__disabled-false states 1`] = `
Object {
  "disabled": false,
  "error": "Something went wrong",
  "valid": undefined,
}
`;

exports[`DatePicker computed usecase-1__disabled-true caretSize 1`] = `"12px"`;

exports[`DatePicker computed usecase-1__disabled-true formattedDate 1`] = `"2018-01-10T00:00:00+00:00"`;

exports[`DatePicker computed usecase-1__disabled-true isEmpty 1`] = `false`;

exports[`DatePicker computed usecase-1__disabled-true states 1`] = `
Object {
  "disabled": true,
  "error": "Something went wrong",
  "valid": undefined,
}
`;
